keyless/circuit-tests/src/arrays.rs

```diff
@@ -0,0 +1,891 @@
+use aptos_keyless_common::input_processing::circuit_input_signals::CircuitInputSignals;
+use aptos_keyless_common::input_processing::config::CircuitPaddingConfig;
+use aptos_keyless_common::input_processing::witness_gen::witness_gen;
+use crate::{TestCircuitHandle};
+use itertools::*;
+use aptos_crypto::poseidon_bn254;
+use ark_bn254::Fr;
+use ark_ff::{Zero, One};
+
+fn build_array_selector_output(len: u32, start: u32, end: u32) -> Vec<u8> {
+    let mut output = Vec::new();
+    for _ in 0..start {
+        output.push(0);
+    };
+    for _ in start..end {
+        output.push(1);
+    };
+    for _ in end..len {
+        output.push(0);
+    };
+    output
+}
+     
+#[test]
+fn array_selector_test() {
+    let circuit_handle = TestCircuitHandle::new("arrays/array_selector_test.circom").unwrap();
+    let out_len = 8;
+    for start in 0..out_len {
+        for end in start+1..=out_len {
+            let output = build_array_selector_output(out_len, start, end);
+            let config = CircuitPaddingConfig::new().max_length("expected_output", out_len as usize);
+            let circuit_input_signals = CircuitInputSignals::new().u64_input("start_index", start as u64).u64_input("end_index", end as u64).bytes_input("expected_output", &output[..]).pad(&config).unwrap();
+            let result = circuit_handle.gen_witness(circuit_input_signals);
+            assert!(result.is_ok());
+        };
+    };
+}
+
+#[test]
+fn array_selector_test_large() {
+    let circuit_handle = TestCircuitHandle::new("arrays/array_selector_test_large.circom").unwrap();
+    let out_len = 2000;
+    let start = 146;
```

Comment:
Any reason why you chose these values?

It might make more sense to iterate over sequential values for start and end, or iterate over random values, rather than just hardcoding arbitrary ones.

Reply:
I iterate over every possible value in `array_selector_test`. This test is intended only to test a large input without being slow to run.